### PR TITLE
Reduce duplicate packing code

### PR DESCRIPTION
**Pull request by billonahill**
Refactors packing impls to do some housekeeping and reduce lines of code:

- Reduces duplicate logic found in the packing algos and their tests
- Adds convenience methods for common logic
- Makes methods that can be private private.
- Reduces verbosity by utilizing `Resource` objects instead of tuples of cpu, ram, disk.

### CODE REVIEW

**Comment by ashvina** [comment]
LGTM, fix failures and Shit it !

File: heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+      containerCpu += (paddingPercentage * containerCpu) / 100;
+      containerRam += (paddingPercentage * containerRam) / 100;
+      containerDiskInBytes += (paddingPercentage * containerDiskInBytes) / 100;
+
+      Resource resource =
+          new Resource(Math.round(containerCpu), containerRam, containerDiskInBytes);
+
+      PackingPlan.ContainerPlan containerPlan =
+          new PackingPlan.ContainerPlan(containerId, instancePlans, resource);
```
Comment (billonahill) [inline heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java]: Resource is still used in the ContainerPlan, but no longer in the PackingPlan.

**Comment by billonahill** [comment]
Thanks @ashvina. @avflor let me know if you'd like time to review.

**Comment by avflor** [comment]
@billonahill I'd like some more time to review this one. Please don't merge yet

**Comment by avflor** [comment]
@billonahill It looks good to me :)

**Pull request merged**